pipeline/steam_extract.py

```diff
@@ -0,0 +1,171 @@
+"""Script to scrape relevant data from the epic games website."""
+
+from os import environ as ENV
+from datetime import datetime, date
+from time import sleep
+from dotenv import load_dotenv
+import requests as req
+from bs4 import BeautifulSoup
+
+
+def get_rating(game_soup: BeautifulSoup) -> float:
+    """Function to get the rating percentage of a game from its URL."""
+
+    rating = game_soup.find('div', class_="review_ctn")
+    if "There are no reviews for this product" in rating:
+        return 0.00
+    else:
+        positive = rating.find('label', {'for': "review_type_positive"})
+        negative = rating.find('label', {'for': "review_type_negative"})
+        if positive:
+            p_num = int(positive.find(
+                'span', class_='user_reviews_count').text.strip('()'))
+
+        else:
+            return 0.00
+        if negative:
+            n_num = int(negative.find(
+                'span', class_='user_reviews_count').text.strip('()'))
+        else:
+            return 100
+
+        return round((p_num/(p_num+n_num)) * 100, 2)
+
+
+def get_platforms(game_soup: BeautifulSoup) -> list:
+    """Function to get the platforms a game is available on from its URL."""
+
+    if game_soup.findAll(
+            'div', class_="sysreq_tab"):
+        platforms = [i.text.strip() for i in game_soup.findAll(
+            'div', class_="sysreq_tab")]
+        for i in range(len(platforms)):
+            if "Linux" in platforms[i]:
+                platforms[i] = "Linux"
+        return platforms
+    else:
+        platforms = game_soup.findAll(
+            'div', class_="sysreq_contents")
+        platform_list = [platform.find(
+            'ul', class_='bb_ul').text for platform in platforms if platform.find(
+                'ul', class_='bb_ul')][0]
+        if "Windows" in platform_list:
+            return "Windows"
+        if "mac" in platform_list:
+            return "macOS"
+        if "Linux" in platform_list:
+            return "Linux"
```

Comment:
Is it possible to have more that one platform returned?

Reply:
The part above handles that so this just catches when theres only one platform